lib/identity_cache/memoized_cache_proxy.rb

```diff
@@ -37,34 +38,58 @@ def with_memoization(&block)
     end
 
     def write(key, value)
-      memoized_key_values[key] = value if memoizing?
-      @cache_fetcher.write(key, value)
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.cache.write', memoizing: memoizing) do
```

Comment:
I'd prefer it if the AS notifications were decorator methods, but I can't think of a clean way to do it that would work with `fetch` and `fetch_multi`.